mage_ai/api/resources/PipelineResource.py

```diff
@@ -432,6 +470,8 @@ def _update_callback(resource):
                         cancel_pipeline_runs(pipeline_runs=pipeline_runs)
                 elif status == 'retry' and pipeline_runs:
                     retry_pipeline_runs(pipeline_runs)
+                elif status == 'retry_incomplete_block_runs':
+                    retry_incomplete_block_runs(pipeline_uuid)
```

Comment:
Shall we support retry incomplete block runs in selected pipeline runs as well? Maybe a TODO

Reply:
Yes, we can do that in a separate story. Retrying all incomplete block runs solves the main issue with tediously going through every pipeline run and retrying across many pages, but we can add this for selected block runs in a future release.